bin/test

```diff
@@ -1,7 +1,24 @@
 #!/bin/bash
-set -euo pipefail
+set -uo pipefail
 
-echo "--- :ruby: Bundle Install"
-bundle install --jobs 4 
-echo "+++ :kubernetes: Run Tests"
-bundle exec rake test TESTOPTS='-v'
\ No newline at end of file
+function print_header() {
+  if [[ ${CI:="0"} == "1" ]]; then
+    echo "+++ :kubernetes: ${1}"
+  else
+    printf "\n\n\033[0;35m⎈   ${1}\033[0m"
+  fi
+}
+
+if [[ ${CI:="0"} == "1" ]]; then
+  echo "--- :ruby: Bundle Install"
+  bundle install --jobs 4
+fi
+
+print_header "Run Unit Tests"
+bundle exec rake unit_test
+
+print_header "Run Non-Parallel Integration Tests"
+bundle exec rake serial_integration_test
+
+print_header "Run Parallel Integration Tests\n"
+PARALLELIZE_ME=1 N=7 bundle exec rake integration_test
```

Comment:
My questions disappeared even though I didn't change this line.

> As for the number, isn't 7 equal to cpus - 1 with our current machine size? (btw I had tried N=8 and saw timeout-related failures on random tests) Also, embarrassing question: how do I detect which to use? if uname == Darwin; else?

Reply:
It is, my thought was we could do `[num_cpus / 2, 1].max` (to account for vcpus). For our purposes the `Darwin` check would work.

This is a nice-to-have suggestion but feel free to merge without it, especially given how fickle the tests have proven with respect to concurrency. The idea behind the suggestion was to avoid having to ever adjust the parallelism values if the machines change.

Reply:
As discussed on Slack, what I just pushed checks if `ENV[PARALLELISM]` is set, and uses that if so. So we can set it and easily tweak it from Buildkite. If it isn't set, it essentially uses num_cpu/2.